src/app/Dashboard/LayoutTemplatePicker.tsx

```diff
@@ -0,0 +1,578 @@
+import { getConfigByName, hasConfigByName } from './Dashboard';
+import { LayoutTemplate, LayoutTemplateContext, LayoutTemplateRecord, recordToLayoutTemplate } from './dashboard-utils';
+import { LayoutTemplateGroup } from './LayoutTemplateGroup';
+import { SearchAutocomplete } from './SearchAutocomplete';
+
+export type LayoutTemplateFilter = 'Suggested' | 'Cryostat' | 'User-submitted';
+
+export type LayoutTemplateSort = 'Name' | 'Card Count'; // TODO: add 'Version' after more version are released
+
+const TemplateSortSelectOption: React.FC<{ sort: LayoutTemplateSort }> = ({ sort }) => {
+  return <SelectOption key={sort} value={sort} />;
+};
+
+const CARD_PREVIEW_LIMIT = 16;
+
+export interface LayoutTemplatePickerProps {
+  onTemplateSelect: (templateName: LayoutTemplate) => void;
+}
+
+export const LayoutTemplatePicker: React.FC<LayoutTemplatePickerProps> = ({ onTemplateSelect }) => {
+  const { selectedTemplate, setSelectedTemplate, setIsUploadModalOpen } = React.useContext(LayoutTemplateContext);
+  const serviceContext = React.useContext(ServiceContext);
+  const dispatch = useDispatch();
+
+  const [searchFilter, setSearchFilter] = React.useState('');
+  const [isFilterSelectOpen, setIsFilterSelectOpen] = React.useState(false);
+  const [selectedFilters, setSelectedFilters] = React.useState<LayoutTemplateFilter[]>([]);
+
+  const [isSortSelectOpen, setIsSortSelectOpen] = React.useState(false);
+  const [selectedSort, setSelectedSort] = React.useState<LayoutTemplateSort | undefined>(undefined);
+  const [sortDirection, setSortDirection] = React.useState<'asc' | 'desc'>('asc');
+
+  const [isDrawerExpanded, setIsDrawerExpanded] = React.useState(false);
+
+  const [isDeleteWarningModalOpen, setIsDeleteWarningModalOpen] = React.useState(false);
+  const [toDelete, setToDelete] = React.useState<string>('');
+
+  const { t } = useTranslation();
+  const recentTemplateRecords: LayoutTemplateRecord[] = useSelector(
+    (state: RootState) => state.dashboardConfigs.templateHistory
+  );
+  const userSubmittedTemplates: LayoutTemplate[] = useSelector(
+    (state: RootState) => state.dashboardConfigs.customTemplates
+  );
+
+  const searchFilteredTemplates = React.useCallback(
+    (templates: LayoutTemplate[]): LayoutTemplate[] => {
+      if (!searchFilter) {
+        return templates;
+      }
+      return templates.filter((t) => t.name.toLowerCase().includes(searchFilter.toLowerCase()));
+    },
+    [searchFilter]
+  );
+
+  // array order is needed for template selection to fallback, if current is deleted
+  const allTemplates: LayoutTemplate[] = React.useMemo(() => {
+    return [BlankLayout, ...CryostatLayoutTemplates, ...userSubmittedTemplates];
+  }, [userSubmittedTemplates]);
+
+  const allSearchableTemplateNames: string[] = React.useMemo(() => {
+    return searchFilteredTemplates(
+      allTemplates.filter((template, index, arr) => arr.findIndex((t) => t.name === template.name) === index)
```

Comment:
What is the filter used for in this case? Seems to return true always?

Reply:
It removes duplicate names since the templates gets mapped to the name.

Reply:
How about using a putting names in a Set? And convert back to Array with Array.from,

Reply:
@maxcao13 ping :)) Using set might be easier to understand, wdu think?